api/v1/controllers/collectors.js

```diff
@@ -138,13 +137,31 @@ function getCollector(req, res, next) {
 /**
  * Update the specified collector's config data. If a field is not included in
  * the querybody, that field will not be updated.
+ * Some fields are only writable by the collector itself. So, if any of those
+ * fields are being updated, check that the token provided in request belongs to
+ * a collector.
  *
  * @param {IncomingMessage} req - The request object
  * @param {ServerResponse} res - The response object
  * @param {Function} next - The next middleware function in the stack
  */
 function patchCollector(req, res, next) {
-  doPatch(req, res, next, helper);
+  let verifyCtrToken = false;
+  for (let i = 0; i < fieldsWritableByCollectorOnly.length; i++) {
```

Comment:
maybe something like https://www.npmjs.com/package/just-intersect ...
e.g. 

```
verifyCtrTkn = intersect(Object.keys(req.body), fieldsWritableByCollectorOnly).length > 0;
```

Reply:
How about this: 
`  // verify controller token if atleast one field is writable by collector
  let verifyCtrToken = false;
  const reqBodyKeys = Object.keys(req.body);
  for (let i = 0; i < fieldsWritableByCollectorOnly.length; i++) {
    const fieldName = fieldsWritableByCollectorOnly[i];
    if (reqBodyKeys.indexOf(fieldName) > -1) {
      verifyCtrToken = true;
      break;
    }
  }`

Same complexity and less dependency on third party package. What do you think?